### PR TITLE
Update for new MSBuildWorkspace

### PR DESCRIPTION
**Pull request by DustinCampbell**
This change updates existing sample that use `MSBuildWorkspace` to the new `Microsoft.CodeAnalysis.Workspaces.MSBuild` NuGet package, and adds a new "SolutionExplorer" sample that significantly exercises `MSBuildWorkspace`.

Because `Microsoft.CodeAnalysis.Workspaces.MSBuild` is not released on https://nuget.org yet, I've added a temporary `NuGet.config` that includes the roslyn MyGet feed. In addition, the package versions used are 2.9.0 beta releases from that feed. Eventually, the `NuGet.config` will go away and the package versions will be updated to stable versions.

### CODE REVIEW

File: samples/CSharp/FormatSolution/FormatSolution.csproj [hunk 1]
```diff
@@ -14,9 +14,11 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Build" Version="15.3.409" />
-    <PackageReference Include="Microsoft.Build.Tasks.Core" Version="15.3.409" />
-    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="2.6.0" />
+    <PackageReference Include="Microsoft.Build.Locator" Version="1.0.13" />
+
+    <!-- These package versions should be updated to 2.9.0 when released on NuGet -->
+    <PackageReference Include="Microsoft.CodeAnalysis.Workspaces.MSBuild" Version="2.9.0-beta4-62810-05" />
```
Comment (Pilchie) [inline samples/CSharp/FormatSolution/FormatSolution.csproj]: Hmm - it looks like we screwed up the versioning of 2.9 already.  We probably don't want these packages to have `-beta4` in them.

@jaredpar @jasonmalinowski ?

Comment (DustinCampbell) [reply samples/CSharp/FormatSolution/FormatSolution.csproj]: I have no idea. Those are the packages on the roslyn feed.

Comment (jasonmalinowski) [reply samples/CSharp/FormatSolution/FormatSolution.csproj]: Ah, cute. Uh, I guess we could bump this back to -beta1 if we wanted to. @brettfo?

Comment (DustinCampbell) [reply samples/CSharp/FormatSolution/FormatSolution.csproj]: what if we need beta4 in 15.8?

**Comment by jmarolf** [comment]
Could we merge these into https://github.com/dotnet/roslyn-sdk/tree/dev15.8.x? I would like samples in master to be against public nuget packages so people can reuse the samples without adding a new feed.

**Comment by DustinCampbell** [comment]
@jmarolf: Sure! I didn't realize we had separate branches for future work.

**Pull request merged**